pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -59,19 +107,32 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
 	if resp.Header.Get("Content-Type") == network.ProtoAcceptContent {
-		return statFromProto(resp.Body)
+		return statFromProto(resp.Body, resp.ContentLength)
 	}
 
 	return statFromPrometheus(resp.Body)
 }
 
-func statFromProto(body io.Reader) (Stat, error) {
+func statFromProto(body io.Reader, l int64) (Stat, error) {
 	var stat Stat
-	bodyBytes, err := ioutil.ReadAll(body)
+	if l <= 0 {
+		return emptyStat, fmt.Errorf("no data received, data size unknown")
```

Comment:
```suggestion
		return emptyStat, errors.New("no data received, data size unknown")
```

No formatting going on.

Reply:
Yeah relic of my cleaning up.